### DISCUSSION TITLE
Feature Request:Enable similarity search providing faiss index

### DISCUSSION THREAD

**Discussion by KIC** [duckdb/duckdb#6894] [Ideas]
I use duckdb for local mainly for development but for data usually bigger than my memory. What I am missing or what would be very beneficial if duckdb would have a similarity search. I was wondering if faiss could be integrated as an index type. Here some pseudo code how I would imagine the index to work.

```
create table kmean_centroids(
  id bigint primary key, 
  center vector[double],  -- could not find an example how to use the vector data type correctly
)

create index centroids on kmean_centroids(center);

select id from kmean_centroids where center=(x, y, z)  -- implicitly uses min distance of index
```

This would also work with individual columns instead of a vector datatype  but it would look a bit odd in the where constraint.

**Comment by Tishj** [comment]
I think you meant to use the `LIST`?
`DOUBLE[]` would be the type in that case
